test/jdk/java/net/httpclient/TimeoutResponseTestSupport.java

```diff
@@ -0,0 +1,413 @@
+            }
+
+        }
+    }
+
+    protected static final ServerRequestPair
+            HTTP1 = ServerRequestPair.of(Version.HTTP_1_1, false),
+            HTTPS1 = ServerRequestPair.of(Version.HTTP_1_1, true),
+            HTTP2 = ServerRequestPair.of(Version.HTTP_2, false),
+            HTTPS2 = ServerRequestPair.of(Version.HTTP_2, true),
+            HTTP3 = ServerRequestPair.of(Version.HTTP_3, true);
+
+    private static SSLContext createSslContext() {
+        try {
+            return new SimpleSSLContext().get();
+        } catch (IOException exception) {
+            throw new UncheckedIOException(exception);
+        }
+    }
+
+    protected record ServerRequestPair(
+            HttpServerAdapters.HttpTestServer server,
+            HttpRequest request,
+            boolean secure) {
+
+        private static final ExecutorService EXECUTOR = Executors.newVirtualThreadPerTaskExecutor();
+
+        private static final CountDownLatch SHUT_DOWN_LATCH = new CountDownLatch(1);
+
+        private static final AtomicInteger SERVER_COUNTER = new AtomicInteger();
+
+        /**
+         * An arbitrary content length to cause the client wait for it.
+         * It just needs to be greater than zero, and big enough to trigger a timeout when delivered slowly.
+         */
+        public static final int CONTENT_LENGTH = 1234;
+
+        public enum ServerHandlerBehaviour {
+            BLOCK_BEFORE_HEADER_DELIVERY,
+            BLOCK_BEFORE_BODY_DELIVERY,
+            DELIVER_BODY_SLOWLY
+        }
+
+        public static volatile ServerHandlerBehaviour SERVER_HANDLER_BEHAVIOUR;
+
+        public static volatile int SERVER_HANDLER_PENDING_FAILURE_COUNT = 0;
+
+        private static ServerRequestPair of(Version version, boolean secure) {
+
+            // Create the server and the request URI
+            SSLContext sslContext = secure ? SSL_CONTEXT : null;
+            String serverId = "" + SERVER_COUNTER.getAndIncrement();
+            HttpServerAdapters.HttpTestServer server = createServer(version, sslContext);
+            server.getVersion();
+            String handlerPath = "/%s/".formatted(CLASS_NAME);
+            String requestUriScheme = secure ? "https" : "http";
+            URI requestUri = URI.create("%s://%s%s-".formatted(requestUriScheme, server.serverAuthority(), handlerPath));
+
+            // Register the request handler
+            server.addHandler(createServerHandler(serverId), handlerPath);
+
+            // Create the request
+            HttpRequest request = createRequestBuilder(requestUri, version).timeout(REQUEST_TIMEOUT).build();
+
+            // Create the pair
+            ServerRequestPair pair = new ServerRequestPair(server, request, secure);
+            pair.server.start();
+            LOGGER.log("Server[%s] is started at `%s`", serverId, server.serverAuthority());
+            return pair;
+
+        }
+
+        private static HttpServerAdapters.HttpTestServer createServer(
```

Comment:
Can we use imports to reduce the long class names please

Reply:
Implemented in 55396846d1a.